.circleci/build_cache

```diff
@@ -0,0 +1,2 @@
+# Change the following version number to trigger a complete rebuild of the doc in CircleCI.
+v2
```

Comment:
haha, neat! Can you add this information to the contributing guide?

Reply:
This is what mne-python does by the way:

```py
export COMMIT_MESSAGE=$(git log --format=oneline -n 1);
if [[ "$COMMIT_MESSAGE" != *"[circle linkcheck]"* ]]
```

etc. Better no?

Reply:
> haha, neat! Can you add this information to the contributing guide?

Sure!

Reply:
> Yeah I saw this but the problem is that CircleCI does not allow flushing existing cache and normal environment variables are invalid in the key specification process...

Okay, let's go with what you suggest here. We can think more about this if there are snags. Normally, if you run `make clean` and `make html` that will clean the directory and then you can save the updated cache. Not sure if you really need to do the md5 hash rather than use a pre-defined key ...

Reply:
`make clean` does no magic more than just `rm -rf *`, which seemed failed to write in the cached dir, according to my previous experiments... Maybe I was wrong, I'll see if I can make a minimal demo later to reproduce this issue.